File: src/main/java/bot/commands/LinksCommand.java
package bot.commands;

import bot.run.AbitVTBot;
import bot.util.CommandsUtil;
import bot.util.Log;
import com.google.common.io.Resources;
import org.json.simple.JSONArray;
import org.json.simple.JSONObject;
import org.json.simple.parser.JSONParser;
import org.json.simple.parser.ParseException;
import org.telegram.telegrambots.meta.api.methods.send.SendMessage;
import org.telegram.telegrambots.meta.api.objects.Message;
import org.telegram.telegrambots.meta.api.objects.Update;
import org.telegram.telegrambots.meta.api.objects.replykeyboard.InlineKeyboardMarkup;
import org.telegram.telegrambots.meta.api.objects.replykeyboard.buttons.InlineKeyboardButton;
import org.telegram.telegrambots.meta.exceptions.TelegramApiException;

import java.io.*;
import java.net.URL;
import java.util.*;

public class LinksCommand implements Command{

    private Log log = new Log();

    private String getTEXT(String chatId){
        ResourceBundle otherBundle = ResourceBundle.getBundle("other", bot.getSql().selectLanguage(chatId).getLocale());
        return otherBundle.getString("links.text");
    }

    private final AbitVTBot bot;

    public LinksCommand(AbitVTBot bot) {
        this.bot = bot;
    }

    @Override
    public void execute(Update update) throws TelegramApiException {

        String chatId = CommandsUtil.getChatId(update);


        SendMessage sendMessage = new SendMessage();
        sendMessage.setChatId(chatId);
        sendMessage.setText(getTEXT(chatId));

        JSONParser parser = new JSONParser();

        InputStream in = getClass().getResourceAsStream("/links.json");
        try (
                BufferedReader reader = new BufferedReader(new InputStreamReader(in));
        )
        {
            JSONArray links = (JSONArray) parser.parse(reader);
            Iterator<JSONObject> iterator = links.iterator();

            List<List<InlineKeyboardButton>> buttons = new ArrayList<>();

            while (iterator.hasNext()){
                JSONObject job = iterator.next();
                InlineKeyboardButton button = new InlineKeyboardButton();
                button.setText((String) job.get("text"));
                button.setUrl((String) job.get("link"));
                List<InlineKeyboardButton> bl = new ArrayList<>();
                bl.add(button);
                buttons.add(bl);
            }

            InlineKeyboardMarkup keyboard = new InlineKeyboardMarkup(buttons);

            sendMessage.setReplyMarkup(keyboard);

            bot.execute(sendMessage);

        } catch (IOException | ParseException e) {
            log.userException(e, update);
        }

    }

    @Override
    public String help(String chatId) {
        ResourceBundle helpBundle = ResourceBundle.getBundle("help", bot.getSql().selectLanguage(chatId).getLocale());
        return helpBundle.getString("links");
    }

    @Override
    public String name() {
        return "/links";
    }

}

File: src/main/java/bot/commands/LinksCommand.kt
package bot.commands

import bot.run.AbitVTBot
import bot.util.CommandsUtil.getChatId
import bot.util.Log
import org.json.simple.JSONArray
import org.json.simple.JSONObject
import org.json.simple.parser.JSONParser
import org.json.simple.parser.ParseException
import org.telegram.telegrambots.meta.api.methods.send.SendMessage
import org.telegram.telegrambots.meta.api.objects.Update
import org.telegram.telegrambots.meta.api.objects.replykeyboard.InlineKeyboardMarkup
import org.telegram.telegrambots.meta.api.objects.replykeyboard.buttons.InlineKeyboardButton
import java.io.BufferedReader
import java.io.IOException
import java.io.InputStreamReader
import java.util.*

class LinksCommand(private val bot : AbitVTBot) : Command {

    private val log: Log = Log()

    override fun execute(update: Update) {
        val chatId = getChatId(update)

        val sendMessage = SendMessage()
        sendMessage.chatId = chatId
        sendMessage.text = bot.resourceOperator.getText("links.text", chatId)!!

        val parser = JSONParser()

        val res = javaClass.getResourceAsStream("/links.json")
        try {
            BufferedReader(InputStreamReader(res!!)).use { reader ->
                val links = parser.parse(reader) as JSONArray
                val buttons: MutableList<List<InlineKeyboardButton>> =
                    ArrayList()
                for (job in links) {
                    job as JSONObject
                    val button = InlineKeyboardButton()
                    button.text = job["text"] as String
                    button.url = job["link"] as String
                    val bl: MutableList<InlineKeyboardButton> = ArrayList()
                    bl.add(button)
                    buttons.add(bl)
                }
                val keyboard = InlineKeyboardMarkup(buttons)
                sendMessage.replyMarkup = keyboard
                bot.execute(sendMessage)
            }
        } catch (e: IOException) {
            log.userException(e, update)
        } catch (e: ParseException) {
            log.userException(e, update)
        }
    }

    override fun help(chatId: String): String {
        return bot.resourceOperator.getHelp("links", chatId)!!
    }

    override fun name(): String {
        return "/links"
    }
}